internal/mode/static/status/prepare_requests.go

```diff
@@ -0,0 +1,335 @@
+package status
+
+import (
+	"fmt"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	v1 "sigs.k8s.io/gateway-api/apis/v1"
+	"sigs.k8s.io/gateway-api/apis/v1alpha2"
+
+	ngfAPI "github.com/nginxinc/nginx-gateway-fabric/apis/v1alpha1"
+	"github.com/nginxinc/nginx-gateway-fabric/internal/framework/conditions"
+	"github.com/nginxinc/nginx-gateway-fabric/internal/framework/helpers"
+	frameworkStatus "github.com/nginxinc/nginx-gateway-fabric/internal/framework/status"
+	staticConds "github.com/nginxinc/nginx-gateway-fabric/internal/mode/static/state/conditions"
+	"github.com/nginxinc/nginx-gateway-fabric/internal/mode/static/state/graph"
+)
+
+// NginxReloadResult describes the result of an NGINX reload.
+type NginxReloadResult struct {
+	// Error is the error that occurred during the reload.
+	Error error
+}
+
+// PrepareRouteRequests prepares status UpdateRequests for the given Routes.
+func PrepareRouteRequests(
+	routes map[types.NamespacedName]*graph.Route,
+	transitionTime metav1.Time,
+	nginxReloadRes NginxReloadResult,
+	gatewayCtrlName string,
```

Comment:
@sjberman 
I used "gatewayCtrlName" because IntelliJ IDE thinks "gatewayCtlrName" has a typo.
I wonder if Ctrl is correct though.
For example, looking at windows keyboards, they use "Ctrl" on the Control button.

Reply:
Yeah, "ctrl" is an abbreviation of "control". Whereas "ctlr" is more of an abbreviation for "controller".